pkgs/applications/misc/noice/default.nix

```diff
@@ -0,0 +1,31 @@
+{ stdenv, fetchgit, ncurses, conf ? null }:
+
+with stdenv.lib;
+
+stdenv.mkDerivation rec {
+  name = "noice-${version}";
+  version = "0.6";
+
+  src = fetchgit {
+    url = "git://git.2f30.org/noice.git";
+    rev = "refs/tags/v${version}";
+    sha256 = "03rwglcy47fh6rb630vws10m95bxpcfv47nxrlws2li2ljam8prw";
+  };
+
+  configFile = optionalString (conf!=null) (builtins.toFile "config.def.h" conf);
+  preBuild = optionalString (conf!=null) "cp ${configFile} config.def.h";
```

Comment:
I've played around with `echo` but -- regardless of all the different kinds of quoting I tried -- somehow it didn't work out. Shouldn't the expression just be as follows?

    preBuild = optionalString (conf!=null) "echo ${conf} > config.def.h";

`conf` itself was originally read by `readFile` in my overlays which should be a string. I guess there is a reason why `toFile` is preferred.